### PR TITLE
Bug 1915968: dkrv2_openshift_secondary_metadata_scraper: check signatures

### PR DESCRIPTION
**Pull request by vrutkovs**
Implement GPG signature check in `dkrv2_openshift_secondary_metadata_scraper` plugin. Use `pgp` crate to fetch signature from HTTP server and decrypt it using a set of public keys. The message should contain digest of the verified image.

Update unit tests to check image signature too. `openshift_secondary_metadata_extraction` is updated with signature config, `test_fixtures` contains:

* `message.json` - json to encrypt
* `vrutkovs.asc` - public key used to sign the message
* `signature-3` - signature generated with `gpg ---sign message.json`.
Its index is set to 3 so that unit test would check signature retries.

`mockito` is running a test HTTP server which serves signature file on temp URL.

Fixes https://issues.redhat.com/browse/OTA-323

### CODE REVIEW

**Comment by vrutkovs** [comment]
/hold

Working on cincinnati-graph-data change to reuse this module, so some minor cleanups may be required

**Comment by vrutkovs** [comment]
/hold cancel

No changes needed yet

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

File: cincinnati/src/plugins/internal/graph_builder/dkrv2_openshift_secondary_metadata_scraper/plugin.rs [hunk 1]
```diff
@@ -156,12 +190,17 @@ impl DkrV2OpenshiftSecondaryMetadataScraperPlugin {
             settings.username = username;
             settings.password = password;
         }
+        let http_client = ClientBuilder::new()
+            .gzip(true)
+            .timeout(Duration::from_secs(DEFAULT_SIGNATURE_FETCH_TIMEOUT_SECS))
+            .build()
+            .context("Building reqwest client")?;
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply cincinnati/src/plugins/internal/graph_builder/dkrv2_openshift_secondary_metadata_scraper/plugin.rs:197]: `reqwest` is http client implementation. We might want to make it generic (i.e. `Building HTTP client`) later on

**Comment by vrutkovs** [comment]
```
 ---- check_slo::_clamp_min_histogram_quantile_0_90_sum_cincinnati_pe_v1_graph_serve_duration_seconds_bucket_by_le_0_5_0_5_ stdout ----
thread 'check_slo::_clamp_min_histogram_quantile_0_90_sum_cincinnati_pe_v1_graph_serve_duration_seconds_bucket_by_le_0_5_0_5_' panicked at 'assertion failed: `(left == right)`
  left: `"0.5751013451262209"`,
 right: `"0.5"`', e2e/tests/slo.rs:55:5 
```
/retest

**Pull request merged**